[CI] FullClusterRestartIT classMethod failing

Similar to https://github.com/elastic/elasticsearch/issues/93504 but now not related to ML.

Marking it for Delivery team due to PR https://github.com/elastic/elasticsearch/pull/93477 that may be related

**Build scan:**
https://gradle-enterprise.elastic.co/s/alt6kcpebfsdk/tests/:x-pack:qa:full-cluster-restart:v8.7.0%23bwcTest/org.elasticsearch.xpack.restart.FullClusterRestartIT

**Reproduction line:**
```
null
```

**Applicable branches:**
8.7

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.restart.FullClusterRestartIT&tests.test=classMethod

**Failure excerpt:**
```
java.lang.RuntimeException: An error occurred orchestrating test cluster.

  at __randomizedtesting.SeedInfo.seed([CEDA55DA1CD14E93]:0)
  at org.elasticsearch.test.cluster.local.LocalClusterHandle.execute(LocalClusterHandle.java:225)
  at org.elasticsearch.test.cluster.local.LocalClusterHandle.writeUnicastHostsFile(LocalClusterHandle.java:206)
  at org.elasticsearch.test.cluster.local.LocalClusterHandle.waitUntilReady(LocalClusterHandle.java:149)
  at org.elasticsearch.test.cluster.local.LocalClusterHandle.start(LocalClusterHandle.java:70)
  at org.elasticsearch.test.cluster.local.LocalElasticsearchCluster$1.evaluate(LocalElasticsearchCluster.java:38)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: java.lang.RuntimeException: Elasticsearch process died while waiting for ports file. See console output for details.

    at org.elasticsearch.test.cluster.local.LocalClusterFactory$Node.lambda$waitUntilReady$0(LocalClusterFactory.java:201)
    at org.elasticsearch.test.cluster.util.Retry.lambda$retryUntilTrue$0(Retry.java:33)
    at org.elasticsearch.test.cluster.util.Retry.lambda$getValueWithTimeout$1(Retry.java:47)
    at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
    at java.lang.Thread.run(Thread.java:833)

```

--- a/test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterFactory.java
+++ b/test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterFactory.java
@@ -132,7 +132,6 @@ public class LocalClusterFactory implements ClusterFactory<LocalClusterSpec, Loc
                 copyExtraConfigFiles();
             }
 
-            deleteGcLogs();
             writeConfiguration();
             createKeystore();
             addKeystoreSettings();
@@ -212,20 +211,6 @@ public class LocalClusterFactory implements ClusterFactory<LocalClusterSpec, Loc
                 throw new RuntimeException("Timed out after " + NODE_UP_TIMEOUT + " waiting for ports files for: " + this);
             } catch (ExecutionException e) {
                 throw new RuntimeException("An error occurred while waiting for ports file for: " + this, e);
-            }
-        }
-
-        private void deleteGcLogs() {
-            try (Stream<Path> logs = Files.list(logsDir)) {
-                logs.filter(l -> l.getFileName().toString().startsWith("gc.log")).forEach(path -> {
-                    try {
-                        IOUtils.deleteWithRetry(path);
-                    } catch (IOException e) {
-                        throw new UncheckedIOException(e);
-                    }
-                });
-            } catch (IOException e) {
-                throw new UncheckedIOException(e);
             }
         }
 
@@ -615,14 +600,13 @@ public class LocalClusterFactory implements ClusterFactory<LocalClusterSpec, Loc
         }
 
         private Map<String, String> getJvmOptionsReplacements() {
-            Path relativeLogsDir = workingDir.relativize(logsDir);
             return Map.of(
                 "-XX:HeapDumpPath=data",
-                "-XX:HeapDumpPath=" + relativeLogsDir,
+                "-XX:HeapDumpPath=" + logsDir,
                 "logs/gc.log",
-                relativeLogsDir.resolve("gc.log").toString(),
+                logsDir.resolve("gc.log").toString(),
                 "-XX:ErrorFile=logs/hs_err_pid%p.log",
-                "-XX:ErrorFile=" + relativeLogsDir.resolve("hs_err_pid%p.log")
+                "-XX:ErrorFile=" + logsDir.resolve("hs_err_pid%p.log")
             );
         }
 

